### PR TITLE
Simulate login through API requests instead of selenium

### PR DESCRIPTION
**Pull request by madhavarshney**
Replace the usage of selenium in `scrape_advanced.py` with a direct implementation that sends POST requests to the necessary FHDA endpoints for SSO.

@phi-line I've opened this PR for you to take an initial look.

TODO:

- [x] Document code

### CODE REVIEW

File: emulate_login.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            # Subsequent HTML pages have an autosubmitting <form>
+            # Emulate auto-submission and use <input> element values for the data
+            data = []
+            for input_el in inputs:
+                if input_el.has_attr('name') and input_el.has_attr('value'):
+                    data.append((input_el.attrs['name'], input_el.attrs['value']))
+
+        i += 1
+
+    return session
```
Comment (madhavarshney) [inline emulate_login.py:72]: May not actually need this line.

File: emulate_login.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        # Convert paths to URLs
+        if next_url.startswith('/'):
+            next_url = last_url.scheme + '://' + last_url.hostname + form.attrs['action']
+
+        if i == 0:
+            # On the first page, use the specified login data
+            # TODO: gracefully exit with error message when environment variables are not present
+            data = [
+                ('j_username', environ['MP_USER']),
```
Comment (phi-line) [inline emulate_login.py:58]: I wonder how easy it is to get a new login that we can use for this API? Just a curiosity

File: selenium_login.py [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# NOTE: With the latest updates to fhda, this script is currently in a broken state
```
Comment (phi-line) [inline selenium_login.py:1]: Just remove this file

**Comment by madhavarshney** [comment]
@phi-line I'm not sure how we would test this, because a username and password is required. Maybe a test that can be run locally?

File: emulate_login.py (hunk 2 above)
Comment (madhavarshney) [reply emulate_login.py:58]: Not sure what you mean... like getting official SSO integration with FHDA?

**Comment by phi-line** [comment]
Yeah, what I was implying here
https://github.com/OpenCourseAPI/OwlAPI/pull/72#discussion_r428765562

is making a dummy My portal account so that we can use it to run tests. I wonder what the shortest path to registration would be to get a MyPortal login.

**Comment by madhavarshney** [comment]
Not sure. We would need to get this service approved first if we were to ask FHDA for an account.

**Comment by phi-line** [comment]
After looking into this the only way to get a MyPortal login is to register as an actual student. So it's definitely off the table unless we get permission from the school.

**Pull request merged**